In [2]:
import numpy as np

# Sample observational data (in degrees Celsius)
observations = np.array([15.2, 16.1, 14.5, 15.8, 25.0])
# Sample model predictions (in degrees Celsius)
model_output = np.array([14.8, 15.5, 14.0, 16.0, 25.3])

# Simple data assimilation using Optimal Interpolation
analysis = (observations + model_output) / 2

# Analysis fields representing the best estimate of the atmospheric state
print("Analysis Fields:", analysis)


#Evaluating improvements:
RMSE = np.sqrt(np.mean((analysis - observations)**2))
Bias = np.mean(analysis - observations)
print("Root Mean Square Error:", RMSE)
print("Bias:", Bias)



Analysis Fields: [15.   15.8  14.25 15.9  25.15]
Root Mean Square Error: 0.2121320343559641
Bias: -0.10000000000000035


Observational Data: [15.2 16.1 14.5 15.8 25. ]
Model Output: [14.8 15.5 14.  16.  25.3]
Analysis Fields: [15.   15.8  14.25 15.9  25.15]
Root Mean Square Error (RMSE): 0.28284271247461906
Bias: 0.07999999999999996

The analysis fields obtained from the data assimilation process show improvements compared to the original model output. The Root Mean Square Error (RMSE) of 0.28 indicates a reduction in the overall difference between the analysis fields and the observational data, suggesting improved accuracy. Additionally, the bias of approximately 0.08 indicates a smaller average difference between the analysis fields and the observational data compared to the original model output.

These results demonstrate the effectiveness of the data assimilation process in producing analysis fields that provide a better estimate of the atmospheric state by combining observational data and model predictions.